runtime/src/eval/macros.rs

```diff
@@ -63,7 +63,12 @@ macro_rules! pop_usize {
 	( $machine:expr, $( $x:ident ),* ) => (
 		$(
 			let $x = match $machine.machine.stack_mut().pop() {
-				Ok(value) => value.as_usize(),
+				Ok(value) if value <= usize::MAX.into() => value.as_usize(),
+				Ok(_) => return Control::Exit(crate::ExitReason::Fatal(
+					crate::ExitFatal::Other(
+						std::borrow::Cow::Borrowed("Number is bigger then usize::MAX"))
```

Comment:
We cannot invent new errors. The EVM must be able to handle these cases gracefully. For example, when geth encounters the EVM code `0x3d3d333e` it does not throw any errors.